### PR TITLE
Windows: Write to debug if a debugger is attached

### PR DESCRIPTION
**Pull request by TheOneRing**
With the recent changes to the logger, Qt creator stopped printing the output if a debugger was attached.

### CODE REVIEW

File: src/libsync/logger.cpp [hunk 1]
```diff
@@ -80,6 +80,12 @@ void Logger::doLog(QtMsgType type, const QMessageLogContext &ctx, const QString
             if (_doFileFlush)
                 _logstream->flush();
         }
+#if defined(Q_OS_WIN)
+        if (IsDebuggerPresent()) {
```
Comment (fmoc) [inline src/libsync/logger.cpp:84]: You may want to extend this check a bit more with other Win32 API debugger checks, e.g., `CheckRemoteDebuggerPresent`.

**Review by fmoc — APPROVED** [review]
Perhaps the debugger checks should be moved into a central static method. The concept could also be extended on other platforms.

**Comment by TheOneRing** [comment]
> 
> 
> Perhaps the debugger checks should be moved into a central static method. The concept could also be extended on other platforms.

This is mainly a workaround for the qt debugger.
The client actually should not behave differently when a debugger is present, but I assume the change here is small enough.

**Comment by fmoc** [comment]
Hence I approved the PR. Just wanted to note that `IsDebuggerPresent` is a bit limited, it only returns "true" when the process is launched by a debugger directly. "Attached" debuggers can be recognized with `CheckRemoteDebuggerPresent` only (at least with the Win32 API).

**Comment by michaelstingl** [comment]
Add a note to the troubleshooting docs?

File: src/libsync/logger.cpp [hunk 2]
```diff
@@ -30,10 +30,24 @@
...
 namespace {
 constexpr int CrashLogSize = 20;
+
+bool isDebuggerPresent()
+{
+#ifdef Q_OS_WIN
+    BOOL debugged;
+    if (!CheckRemoteDebuggerPresent(GetCurrentProcess(), &debugged)) {
+        const auto error = GetLastError();
+        qDebug() << "Failed to dtect debugger" << QString::fromWCharArray(_com_error(error).ErrorMessage());
```
Comment (fmoc) [inline src/libsync/logger.cpp:45]: `detect`

**Pull request merged**